src/ansys/fluent/core/quantity/dimensions.py

```diff
@@ -0,0 +1,123 @@
+import ansys.fluent.core.quantity as q
+
+
+class Dimensions(object):
+    """Initialize a Dimensions object using a dimensions list or a unit string.
+
+    Parameters
+    ----------
+    units: str
+        Unit string of quantity.
+    dimensions: list
+        List of dimensions.
+
+    Returns
+    -------
+    Dimensions instance.
+    """
+
+    def __init__(
+        self, units: str = None, dimensions: list = None, unit_sys: str = None
+    ):
+        self._units_table = q.UnitsTable()
+        unit_sys = unit_sys or "SI"
+        units = units or " "
+
+        if units:
+            self._unit = units
+            self._dimensions = self._units_to_dim(units=units)
+
+        if dimensions:
+            self._dimensions, self._unit = self._dim_to_units(
+                dimensions=dimensions, unit_sys=unit_sys
+            )
+
+    def _dim_to_units(self, dimensions: list, unit_sys: str) -> str:
+        """Convert a dimensions list into a unit string.
+
+        Parameters
+        ----------
+        dimensions : list
+            List of unit dimensions.
+
+        unit_sys : str
+            Unit system of dimensions.
+
+        Returns
+        -------
+        units : str
+            Unit string representation of dimensions.
+        """
+        # Ensure dimensions list contains 9 terms
+        dimensions = [float(dim) for dim in dimensions + ((9 - len(dimensions)) * [0])]
+        sys_order = {
```

Comment:
ideally not embedded inside a method

Reply:
Added predefined unit systems to the yaml config file. If any new systems need to be embedded they can be added directly to the config file and work with the exiting quantity implementation. 

Not sure about adding a UnitSystems class at the moment. If one of the objectives of this quantity module is to allow users to define custom unit systems, then adding a UnitSystems class would definitely be effective. For now, the unit systems are just small dictionaries so I've added them to the config file.

https://github.com/ansys/pyfluent/blob/911b052f1e7162a45f2bc72965029fcf5045470b/src/ansys/fluent/core/quantity/quantity_config.yaml#L243-L279